### PR TITLE
Test case and CI/CD setup

### PR DESCRIPTION
**Pull request by jinalviranii**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,29 +3,28 @@
...
+    "test:vue": "jest --updateSnapshot --projects src --no-cache",
+    "test": "npm run test:node && npm run test:vue"
   },
   "dependencies": {
     "@gofynd/fdk-client-javascript": "^1.3.5",
     "@gofynd/nitrozen-vue": "0.0.33",
     "axios": "^0.27.2",
+    "axios-mock-adapter": "^1.21.2",
     "body-parser": "^1.20.0",
     "convict": "^6.2.3",
```
Comment (vivek-gofynd) [inline package.json]: @jinalviranii please remove dependency of convict

File: app/fdk/index.js [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
         uninstall: async (req) => {
             // Write your code here to cleanup data related to extension
             // If task is time taking then process it async on other process.
         }
     },
-    storage: new RedisStorage(appRedis, "exapmple-fynd-platform-extension"), // add your prefix
+    storage: new RedisStorage(appRedis,"exapmple-fynd-platform-extension"), // add your prefix
     access_mode: "offline",
-    cluster: config.extension.fp_api_server// this is optional by default it points to "https://api.fynd.com".
+    cluster:  config.extension.fp_api_server // this is optional (default: "https://api.fynd.com")
```
Comment (vivek-gofynd) [inline app/fdk/index.js]: If this is optional can we remove this from boilerplate @jinalviranii

File: app/server.js [hunk 3]
```diff
@@ -1,40 +1,59 @@
...
+// Middleware to parse cookies with a secret key
 app.use(cookieParser("ext.session"));
+
+// Middleware to parse JSON bodies with a size limit of 2mb
 app.use(bodyParser.json({
     limit: '2mb'
 }));
+
+// Serve environment variables as a JavaScript file
 app.get('/env.js', (req, res) => {
```
Comment (vivek-gofynd) [inline app/server.js]: Is this required @jinalviranii ?

File: app/server.js [hunk 4]
```diff
@@ -1,40 +1,59 @@
...
 app.use("/", fdkExtension.fdkHandler);
+
+// FDK extension api route which has auth middleware and FDK client instance attached to it.
 const apiRoutes = fdkExtension.apiRoutes;
-apiRoutes.use('/v1.0', v1Router);
+apiRoutes.use('/v1.0', productRouter);
 app.use('/api', apiRoutes);
 
+// Serve the Vue app for company-specific routes
+app.get('/company/:company_id', (req, res) => {
```
Comment (vivek-gofynd) [inline app/server.js]: Will this be required @jinalviranii

File: package.json [hunk 5]
```diff
@@ -3,29 +3,28 @@
...
-    ]
+    "test:node": "jest --config app/jest.config.js app --no-cache --json --outputFile=./coverage/coverage.json",
+    "test:vue": "jest --updateSnapshot --projects src --no-cache",
+    "test": "npm run test:node && npm run test:vue"
   },
   "dependencies": {
     "@gofynd/fdk-client-javascript": "^1.3.5",
     "@gofynd/nitrozen-vue": "0.0.33",
     "axios": "^0.27.2",
+    "axios-mock-adapter": "^1.21.2",
```
Comment (vivek-gofynd) [inline package.json]: @jinalviranii can we move this to devdependency ?

File: package.json [hunk 6]
```diff
@@ -3,29 +3,28 @@
...
     "body-parser": "^1.20.0",
     "convict": "^6.2.3",
     "cookie-parser": "^1.4.6",
     "core-js": "^3.25.2",
     "dotenv": "^10.0.0",
     "express": "^4.18.1",
     "fdk-extension-javascript": "git+https://github.com/gofynd/fdk-extension-javascript.git#v0.6.0-beta.3",
+    "flush-promises": "^1.0.2",
```
Comment (vivek-gofynd) [inline package.json]: dev-dependency @jinalviranii ?

File: src/pages/Home.vue [hunk 7]
```diff
@@ -96,6 +96,7 @@ export default {
     return {
       product_list: [],
       pageLoading: false,
+      moonlight_URL: window.env?.fp_api_server || 'https://api.fynd.com'
```
Comment (vivek-gofynd) [inline src/pages/Home.vue]: Add static URL of production only @jinalviranii

Comment (vivek-gofynd) [reply src/pages/Home.vue]: Use generic name like documentation_url @jinalviranii

File: src/pages/Home.vue [hunk 8]
```diff
@@ -104,12 +105,12 @@ export default {
   },
   computed: {
     getDocumentPageLink() {
```
Comment (vivek-gofynd) [inline src/pages/Home.vue:106]: Check if this is really required @jinalviranii

File: src/pages/Home.vue [hunk 9]
```diff
@@ -104,12 +105,12 @@ export default {
...
-      return "https://api.uat.fyndx1.de"
+      return this.moonlight_URL
         .replace("api", "partners")
         .concat(this.isApplicationLaunch ? DOC_APP_URL_PATH : DOC_URL_PATH);
     },
     isApplicationLaunch() {
```
Comment (vivek-gofynd) [inline src/pages/Home.vue:111]: @jinalviranii Is this required for company level extension ?

**Review by vivek-gofynd — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (jinalviranii) [reply package.json]: @vivek-gofynd Why we want to remove this? We discussed to not remove if you remember

File: app/server.js (hunk 3 above)
Comment (jinalviranii) [reply app/server.js]: Yes this is required to read fp_api_server in frontend component

File: package.json (hunk 5 above)
Comment (jinalviranii) [reply package.json]: I am using this to mock apis. any use case to remove this package?

File: app/fdk/index.js (hunk 2 above)
Comment (jinalviranii) [reply app/fdk/index.js]: Removed

File: package.json (hunk 6 above)
Comment (jinalviranii) [reply package.json]: Moved to dev dependency

File: app/server.js (hunk 3 above)
Comment (jinalviranii) [reply app/server.js]: Also if we want static documentation link then we do not need this, but for boilerplate do we need to share how end user can access env on client side? @vivek-gofynd

File: src/pages/Home.vue (hunk 7 above)
Comment (jinalviranii) [reply src/pages/Home.vue]: Used production documentation static link

File: src/pages/Home.vue (hunk 8 above)
Comment (jinalviranii) [reply src/pages/Home.vue:106]: Yes this is needed

File: src/pages/Home.vue (hunk 9 above)
Comment (jinalviranii) [reply src/pages/Home.vue:111]: Yes required for both company and application level extension as per figma provided by deepu

File: app/server.js (hunk 4 above)
Comment (jinalviranii) [reply app/server.js]: Not required, Was added for react boilerplate, hence kept same

File: package.json (hunk 5 above)
Comment (vivek-gofynd) [reply package.json]: Move it to dev-dependency

File: app/fdk/index.js [hunk 10]
```diff
@@ -1,31 +1,29 @@
...
-
+        
         uninstall: async (req) => {
             // Write your code here to cleanup data related to extension
             // If task is time taking then process it async on other process.
         }
     },
-    storage: new RedisStorage(appRedis, "exapmple-fynd-platform-extension"), // add your prefix
+    storage: new RedisStorage(appRedis,"exapmple-fynd-platform-extension"), // add your prefix
     access_mode: "offline",
```
Comment (vivek-gofynd) [inline app/fdk/index.js]: Use online mode here

File: package.json [hunk 11]
```diff
@@ -3,29 +3,28 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "start": "node index.js",
-    "serve": "vue-cli-service serve",
+    "start": "nodemon index.js",
+    "serve": "PORT=8081 vue-cli-service serve",
+    "dev-start": "concurrently \"npm run start\" \"npm run serve\"",
```
Comment (vivek-gofynd) [inline package.json:8]: @jinalviranii dev-start command should support HMR on the same port on which backend server is running

File: app/server.js [hunk 12]
```diff
@@ -1,40 +1,59 @@
...
+  res.sendFile(path.resolve(__dirname, "../dist/index.html"));
+});
+
+// Serve the Vue app for all other routes
 app.get('*', (req, res) => {
     res.contentType('text/html');
-    res.sendFile(path.join(__dirname, '../', 'dist/index.html'))
+    res.sendFile(path.resolve(__dirname, '../dist/index.html'));
```
Comment (vivek-gofynd) [inline app/server.js]: Use path.join here

File: app/server.js (hunk 4 above)
Comment (jinalviranii) [reply app/server.js]: Removed

**Review by vivek-gofynd — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (jinalviranii) [reply package.json]: Done

File: app/fdk/index.js (hunk 10 above)
Comment (jinalviranii) [reply app/fdk/index.js]: Used online mode

File: app/server.js (hunk 12 above)
Comment (jinalviranii) [reply app/server.js]: Done

**Pull request closed without merging**